AISKU/src/Initialization.ts

```diff
@@ -411,3 +427,49 @@ export class Initialization implements IApplicationInsights {
         _this.config.diagnosticLogInterval && _this.config.diagnosticLogInterval > 0 ? _this.config.diagnosticLogInterval : 10000;
     }
 }
+
+// tslint:disable-next-line
+(function () {
+    let sdkSrc = null;
+    let isModule = false;
+
+    try {
+        // Try and determine whether the sdk is being loaded from the CDN
+        // currentScript is only valid during initial processing
+        let scrpt = (document ||{} as any).currentScript;
+        if (scrpt) {
+            sdkSrc = scrpt.src;
+        // } else {
+        //     // We need to update to at least typescript 2.9 for this to work :-(
+        //     // Leaving as a stub for now so after we upgrade this breadcrumb is available
+        //     let meta = import.meta;
+        //     sdkSrc = (meta || {}).url;
+        //     isModule = true;
+        }
+    } catch (e) {
+    }
+
+    if (sdkSrc) {
+        try {
+            let url = sdkSrc.toLowerCase();
+            if (url) {
+                let src = "";
+                if (url.indexOf("://az416426.vo.msecnd.net/") !== -1) {
+                    src = "cdn1";
+                    if (url.indexOf("/scripts/") === -1) {
+                        if (url.indexOf("/next/") !== -1) {
+                            src += "-next";
+                        } else if (url.indexOf("/beta/") !== -1) {
+                            src += "-beta";
+                        }
+                    }
+                }
+
+                if (src) {
+                    _internalSdkSrc = src + (isModule ? ".mod" : "");
+                }
+            }
+        } catch (e) {
+        }
```

Comment:
When caught, does that mean NPM? Is it possible to know NPM vs CDN vs unknown?

Reply:
This catch is unexpected and added purely to ensure that we don't crash, in which case this will mean that we don't know the outcome (same as what would happen with all IE and really old browsers).

So it's not possible to "always" determine npm, cdn or unknown. This is also why I added the snippet property as it should always be defined in which case we could assume that snippet users are using the primary CDN (vs their own).

Reply:
ok so for now it will report primary CDN or (NPM + custom CDN + unknown)?

Reply:
It will report primary CDN or nothing.
And when nothing we can assume that its NPM if also there is no snippet.